vespalib/src/vespa/vespalib/util/array.hpp

```diff
@@ -113,24 +113,24 @@ void Array<T>::resize(size_t n)
         reserve(n);
     }
     if (n > _sz) {
-        construct(array(_sz), n-_sz, std::tr1::has_trivial_destructor<T>());
+        construct(array(_sz), n-_sz, std::is_trivially_default_constructible<T>());
     } else if (n < _sz) {
-        std::_Destroy(array(n), array(_sz));
+        std::destroy(array(n), array(_sz));
     }
     _sz = n;
 }
 
 template <typename T>
-void move(T * dest, T * source, size_t sz, std::tr1::false_type)
+void move(T * dest, T * source, size_t sz, std::false_type)
 {
     for (size_t i(0); i < sz; i++) {
-        std::_Construct(dest + i, std::move(*(source + i)));
-        std::_Destroy(source + i);
+        ::new (static_cast<void *>(dest + i)) T(std::move(*(source + i)));
+        std::destroy_at(source + i);
```

Comment:
Out of scope for this PR, but for future consideration: since this bulk move is not exception safe, it would be useful to predicate moving on `std::is_nothrow_move_constructible<T>` to ensure that moves can never leave the array in an inconsistent state if an exception is thrown.